README.md

```diff
@@ -349,27 +330,39 @@ documentation](https://docs.rubocop.org/rubocop/configuration.html#inheriting-co
   names.map(&:upcase)
   ~~~
 
-* Prefer `{...}` over `do...end` for single-line blocks. Avoid using `{...}` for
-  multi-line blocks. Always use `do...end` for "control flow" and "method
-  definitions" (e.g. in Rakefiles and certain DSLs). Avoid `do...end` when
-  chaining.
+* Prefer `{...}` over `do...end` for single-line blocks.
+
+* Prefer `do..end` over `{...}` for multi-line blocks.
+
+* Omit `return` where possible.
 
-* Avoid `return` where not required for control flow.
+* Omit `self` where possible.
 
-* Avoid `self` where not required (it is only required when calling a self write
-  accessor).
+  ~~~ ruby
+  # bad
+  self.my_method
+
+  # good
+  my_method
+
+  # also good
+  attr_writer :name
+
+  def my_method
+    self.name = 'Rafael' # `self` is neeeded to reference the attribute writer.
+  end
+  ~~~
 
-* Using the return value of = in a conditional expression is okay if wrapped in
-  parenthesis
+* Wrap assignment in parentheses when using its return value in a conditional
+  statement.
 
   ~~~ ruby
-  if (v = /foo/.match(string))
+  if (value = /foo/.match(string))
   ~~~
 
 * Use `||=` to initialize variables only if they're not already initialized.
 
-* Don't use `||=` to initialize boolean variables (consider what would happen if
-  the current value happened to be `false`).
+* Avoid using `||=` to initialize boolean variables.
```

Comment:
Might be worth extending this from just booleans to also values that could be `nil`.

Reply:
it _should_ extend to `nil` as `||` triggers that way for both `false` and `nil` so that the computation would be triggered again

Reply:
Yeah. Consider

```ruby
def value
  @value ||= sleep(1000) && ['string', true, false, nil].shuffle
end
```

The truthy cases will be memoized, but the falsey ones won't be and would duplicate the computation, whereas this would memoize properly the first time:

```ruby
def value
  return @value if defined?(@value)
  @value = sleep(1000) && [true, false, nil].shuffle
end
```

Really, the rule should be

> Do not use `||=` to **memoize** potentially falsey variables (`false` or `nil`).
> &lt;code example demonstrating use of defined?&gt;

If the use case is potentially falsey memoization, then you **never** want to use `||=`. If you're using `||=` for something else, it **might** be fine.